File: LEETCODE_2924.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        unordered_map<int,int> mp;
        for(vector<int>& edge: edges) {
            mp[edge[1]]++;
        }
        int winner = -1, winner_count = 0;
        for(int i = 0; i < n; i++) {
            if(mp[i] == 0) {
                winner = i;
                winner_count++;
            }
        }
        return (winner_count == 1) ? winner : -1;
    }
};

File: LEETCODE_2924.java
import java.util.*;

class Solution {
    public int findChampion(int n, int[][] edges) {
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i = 0; i < n; i++) {
            mp.put(i,1);
        }
        for(int i = 0; i < edges.length; i++) {
            mp.put(edges[i][1], 0);
        }
        int answer = -1, winnerCount = 0;
        for(int i = 0; i < n; i++) {
            if(mp.get(i) == 1) {
                answer = i;
                winnerCount++;
            }
        }
        return (winnerCount == 1) ? answer : -1;
    }
}